crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -119,6 +119,27 @@ impl BlobVec {
         }
     }
 
+    /// Reserves the minimum capacity for at least `additional` more elements to be inserted in the given `BlobVec`.
+    #[inline]
+    pub fn reserve(&mut self, additional: usize) {
+        if self.capacity - self.len < additional {
+            self.do_reserve(additional);
+        }
+    }
+
+    /// Similar to `reserve_exact`. This method ensures that the capacity will grow at least `self.capacity()` if there is no
+    /// enough space to hold `additional` more elements.
+    #[cold]
+    fn do_reserve(&mut self, additional: usize) {
```

Comment:
I'd follow the style the Rust project uses for their Vec implementation, and scope the cold function within the reserve function itself: https://github.com/rust-lang/rust/blob/6bc08a725f888a06ea3c6844f3d0cc2d2ebc5142/library/alloc/src/raw_vec.rs#L294.